packages/dev/core/src/Rendering/IBLShadows/iblShadowsVoxelTracingPass.ts

```diff
@@ -254,76 +283,107 @@ export class _IblShadowsVoxelTracingPass {
+            "voxelTracingPass",
+            {
+                width: this._engine.getRenderWidth(),
+                height: this._engine.getRenderHeight(),
+            },
+            "iblShadowVoxelTracing",
+            this._scene,
+            textureOptions,
+            false,
+            false,
+            Constants.TEXTURETYPE_UNSIGNED_INT
+        );
+        this._outputTexture.refreshRate = -1;
+        this._outputTexture.autoClear = false;
+        this._outputTexture.defines = defines;
+        // Need to set all the textures first so that the effect gets created with the proper uniforms.
+        this._update(this._scene.activeCamera!);
+
+        let counter = 0;
+        this._scene.onBeforeRenderObservable.add(() => {
+            counter = 0;
+        });
+        this._scene.onAfterRenderTargetsRenderObservable.add(() => {
+            if (++counter == 2) {
+                if (this.enabled && this._outputTexture.isReady()) {
+                    this._update(this._scene.activeCamera!);
+                    this._outputTexture.render();
+                }
+            }
         });
     }
 
-    private _updatePostProcess(effect: Effect, camera: Camera) {
+    private _update(camera: Camera) {
         if (this._scene.useRightHandedSystem) {
-            effect.defines = "#define RIGHT_HANDED\n";
+            this._outputTexture.defines = "#define RIGHT_HANDED\n";
         }
-        effect.setMatrix("viewMtx", camera.getViewMatrix());
-        effect.setMatrix("projMtx", camera.getProjectionMatrix());
+        this._outputTexture.setMatrix("viewMtx", camera.getViewMatrix());
+        this._outputTexture.setMatrix("projMtx", camera.getProjectionMatrix());
         camera.getProjectionMatrix().invertToRef(this._cameraInvProj);
         camera.getViewMatrix().invertToRef(this._cameraInvView);
-        effect.setMatrix("invProjMtx", this._cameraInvProj);
-        effect.setMatrix("invViewMtx", this._cameraInvView);
-        effect.setMatrix("wsNormalizationMtx", this._invWorldScaleMatrix);
+        this._outputTexture.setMatrix("invProjMtx", this._cameraInvProj);
+        this._outputTexture.setMatrix("invViewMtx", this._cameraInvView);
+        this._outputTexture.setMatrix("wsNormalizationMtx", this._invWorldScaleMatrix);
 
         this._frameId++;
 
-        const downscaleSquared = this._downscale * this._downscale;
+        // const downscaleSquared = this._downscale * this._downscale;
         let rotation = this._scene.useRightHandedSystem ? -(this._envRotation + 0.5 * Math.PI) : this._envRotation - 0.5 * Math.PI;
         rotation = rotation % (2.0 * Math.PI);
-        effect.setVector4("shadowParameters", new Vector4(this._sampleDirections, this._frameId / downscaleSquared, this._downscale, rotation));
-        const offset = new Vector2(0.0, 0.0);
+        this._outputTexture.setVector4("shadowParameters", new Vector4(this._sampleDirections, this._frameId, 1.0, rotation));

```

Comment:
Same as above regarding the creation of objects and GC.

Reply:
Done